File: data-base/teachers/put.js
const { MongoClient } = require('mongodb');

// URI de conexión a MongoDB (asegúrate de reemplazarlo con el tuyo)
const uri = 'mongodb+srv://alan11gt:ioUvPgAvDZcVwWXs@cluster0.2b8il.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0';
const client = new MongoClient(uri);

async function run() {
  try {
    // Conectar a MongoDB
    await client.connect();
    console.log('Conectado a MongoDB correctamente');

    // Seleccionar la base de datos y colección
    const database = client.db('ChatbotDB'); // Reemplaza con el nombre de tu base de datos
    const collection = database.collection('teachers register'); // Colección "teachers register"

    // **Actualizar un Solo Documento**
    // Ejemplo: actualizar el 'Email' del documento donde 'Nombre' sea 'Dr. Juan Pérez'
    const filter = { Nombre: 'Dr. Juan Pérez' }; // Criterio de búsqueda para encontrar el documento
    const updateDoc = {
      $set: {
        Email: 'juan.perez.actualizado@uabc.edu.mx', // Nuevo valor para el campo 'Email'
      },
    };

    const result = await collection.updateOne(filter, updateDoc);
    console.log(`Documentos actualizados: ${result.modifiedCount}`);
  } catch (err) {
    console.error('Error al conectar a MongoDB:', err);
  } finally {
    // Cerrar la conexión con MongoDB
    await client.close();
  }
}

run().catch(console.dir);

File: data-base/users/put.py
from pymongo import MongoClient
import re

# Conexión a MongoDB
client = MongoClient("mongodb+srv://alan11gt:ioUvPgAvDZcVwWXs@cluster0.2b8il.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0")
db = client["ChatbotDB"]
usuarios_collection = db["users"]

# Mostrar todos los nombres en la base de datos
def listar_nombres():
    usuarios = usuarios_collection.find({}, {"Nombre": 1, "_id": 0})  # Solo obtener el campo "Nombre"
    print("Nombres en la base de datos:")
    for usuario in usuarios:
        print(usuario["Nombre"])

# Ejemplo de uso
if __name__ == "__main__":
    listar_nombres()

# Buscar usuario por nombre
def buscar_usuario_por_nombre(nombre):
    nombre = nombre.strip()  # Eliminar espacios en blanco
    print(f"Buscando usuario con nombre exacto: '{nombre}'")  # Depuración
    usuario = usuarios_collection.find_one({"Nombre": nombre})  # Búsqueda exacta sin regex
    if usuario:
        print(f"Usuario encontrado: {usuario}")
        return usuario
    else:
        print("Usuario no encontrado para búsqueda.")
        return None

# Ejemplo de uso
if __name__ == "__main__":
    buscar_usuario_por_nombre("Gokú Actualizado")

# Actualizar el documento para corregir un nombre
def corregir_usuario_por_nombre(nombre_incorrecto, nuevo_nombre):
    usuario = buscar_usuario_por_nombre(nombre_incorrecto)
    if usuario:
        resultado = usuarios_collection.update_one({"Nombre": usuario["Nombre"]}, {"$set": {"Nombre": nuevo_nombre}})
        if resultado.matched_count > 0:
            print(f"Usuario actualizado correctamente a: {nuevo_nombre}")
        else:
            print("Error al intentar actualizar el usuario.")
    else:
        print("No se encontró el usuario para actualizar.")

# Ejemplo de uso
if __name__ == "__main__":
    # Corregir un nombre de usuario
    corregir_usuario_por_nombre("Goku Ramirez", "Goku Son")